lib/files.c

```diff
@@ -789,3 +789,92 @@ tool_rc files_tpm2b_attest_to_tpms_attest(TPM2B_ATTEST *quoted, TPMS_ATTEST *att
 
     return tool_rc_success;
 }
+
+tool_rc files_load_unique_data(const char *file_path,
+TPM2B_PUBLIC *public_data) {
+
+    UINT16 unique_size = UINT16_MAX;
+    uint8_t file_data[UINT16_MAX];
+    bool result = files_load_bytes_from_buffer_or_file_or_stdin(NULL,
+    file_path, &unique_size, file_data);
+    if (!result) {
+        LOG_ERR("Failed to load unique data from file/ stdin.");
+        return tool_rc_general_error;
+    }
+
+    /*
+     * If the data is specified as a file, the user is responsible for ensuring
+     * that that this buffer is formatted per TPMU_PUBLIC_ID union.
+     */
+    if (file_path) {
+        memcpy(&public_data->publicArea.unique, file_data, unique_size);
+        return tool_rc_success;
+    }
+
+    /*
+     * What follows is all cases of unstructured data specified on the stdin
+     * that is to be used as the unique data.
+     */
+
+    /*
+     * TPM2_MAX_RSA_KEY_BYTES which expands to 512 bytes is the maximum unique
+     * size for RSA and other key types.
+     *
+     * Additionally this may still prove to be a bigger value than what has been
+     * implemented in a TPM. For example the value of MAX_RSA_KEY_BYTES is 256
+     * for ibmSimulator as specified in implementation.h
+     */
+    if (unique_size > TPM2_MAX_RSA_KEY_BYTES) {
+        LOG_ERR("Unique data too big for any object's allowed unique size.");
+        return tool_rc_general_error;
+    }
```

Comment:
Superfluous check— unique_size max is initialized with TPM2_MAX_RSA_KEY_BYTES and when updated never exceeds its initialization value.